src/software-management/self-update.md

```diff
@@ -0,0 +1,101 @@
+
+# Introduction
+
+
+Thin-edge self-update allows a Cloud Operator to update one thin-edge version that is installed on a device to a newer version, via the Cloud connection. The feature is referred as self-update cause that update process is managed by thin-edge own's Software Management.
+
+# Scope
+  * The self-update makes use of usual thin-edge Software Management plugins (e.g. "APT"). So is not absolutely fail-safe, instead of that it is the foundation for it.<br/>
+    For absolutely fail-safe self-update the device would need to provide and use some transaction-safe package manager and some powerfail-safe filesystem.
+     
+  * Consequences: 
+     * Self-update can fail and could left the system in an unstable state. The device might be disconnected from Cloud.
+     * Potential fail reasons (without claim of completeness):
+       * No rollback of already processes package of one self-update request, when some subsequent package of same request fails.
+         <br/>-> leads to partially executed update. 
+       * Power-fail during update.
+         <br/>-> leads to partially executed update
+         <br/>-> can corrupt the package currently about to be updated
+         <br/>-> can corrupt the filesystem
+       * Installation of incompatibel set of thin-edge package versions (e.g. SM Agent of one release, and Mapper of another release).
+         <br/>-> thin-edge function might not given, when signifcant change happened between both release
+       * Tool(s) required by thin-edge installation procedure missing on the device (e.g. 'grep').
+         <br/>-> according package installation will stop.
+         
+  * Measures for more robusness:
+    * There are plans for thin-edge packages to reduce risks for installation failures:
+      <br/>TODO: For both, add some reference (some ‚ticket, discussion, ...)
+      1. Providing one thin-edge core package that contains everything needed for self-update (assumedly "Mapper", "SM Agent" and "SM Plugin"). 
+          So self-update just needs that one-and-only package; all other thin-edge packages can be update via usual Software Management.
+      2. Avoiding use of external tools in package installation logic (move from Shell scripts to Rust binary).
+    * If the custom device provides some transaction-safe package manager and some powerfail-safe filesystem, the device owner can 
+      provide an according Software Management plugin, and can make use of that by processing a self-update on the according module type. 
+
+*Hint: Before executing self-update on devices in the field it shall be tested on a kind of golden device that equals the device(s) in the field in system-environment, configuration and thin-edge version.*
+
+
+
+
+# Execution Flow and involved Components
+
+![Sequence Diagram Update SW-list](images/self-update.drawio.svg)
+
+
+# Design Principles
+
+* Thin-edge updates are managed by a specific software component named `self-updater`.
```

Comment:
When the actual binary is created, it'd be better if it's called `tedge-updater` or even `tedge-self-updater` to avoid any ambiguities on what that `self` is.

Reply:
In the current ingoing implementation the binary is alread prefixed with `tedge`. So lets add that to the spec:

```suggestion
* Thin-edge updates are managed by a specific software component named tedge-self-updater (in following sections referenced as `self-updater`).
```